extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/model/predictor/FilterService.java

```diff
@@ -0,0 +1,124 @@
+package org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor;
+
+import ai.onnxruntime.OnnxTensor;
+import ai.onnxruntime.OnnxValue;
+import ai.onnxruntime.OrtException;
+import ai.onnxruntime.OrtSession;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.data.ThrottlingMessage;
+import org.springframework.util.CollectionUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class FilterService {
+
+    public Map<String, Map<String, Boolean>> filterBidders(
+            OnnxModelRunner onnxModelRunner,
+            List<ThrottlingMessage> throttlingMessages,
+            Double threshold) {
+
+        final OrtSession.Result results;
+        try {
+            final String[][] throttlingInferenceRows = convertToArray(throttlingMessages);
+            results = onnxModelRunner.runModel(throttlingInferenceRows);
+            return processModelResults(results, throttlingMessages, threshold);
+        } catch (OrtException e) {
+            throw new PreBidException("Exception during model inference: ", e);
+        }
+    }
+
+    private static String[][] convertToArray(List<ThrottlingMessage> messages) {
+        return messages.stream()
+                .map(message -> new String[]{
+                        message.getBrowser(),
+                        message.getBidder(),
+                        message.getAdUnitCode(),
+                        message.getCountry(),
+                        message.getHostname(),
+                        message.getDevice(),
+                        message.getHourBucket(),
+                        message.getMinuteQuadrant()})
+                .toArray(String[][]::new);
+    }
+
+    private Map<String, Map<String, Boolean>> processModelResults(
+            OrtSession.Result results,
+            List<ThrottlingMessage> throttlingMessages,
+            Double threshold) {
+
+        validateThrottlingMessages(throttlingMessages);
+
+        return StreamSupport.stream(results.spliterator(), false)
+                .filter(onnxItem -> {
+                    validateOnnxTensor(onnxItem);
+                    return Objects.equals(onnxItem.getKey(), "probabilities");
+                })
+                .map(onnxItem -> (OnnxTensor) onnxItem.getValue())
+                .map(tensor -> {
+                    validateTensorSize(tensor, throttlingMessages.size());
+                    return extractAndProcessProbabilities(tensor, throttlingMessages, threshold);
+                })
+                .flatMap(map -> map.entrySet().stream())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
```
        return StreamSupport.stream(results.spliterator(), false)
                .peek(FilterService::validateOnnxTensor)
                .filter(onnxItem -> Objects.equals(onnxItem.getKey(), "probabilities"))
                .map(OnnxItem::getValue)
                .map(OnnxTensor.class::cast)
                .peek(tensor -> validateTensorSize(tensor, throttlingMessages.size()))
                .map(tensor -> extractAndProcessProbabilities(tensor, throttlingMessages, threshold))
                .map(Map::entrySet)
                .flatMap(Collection::stream)
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Reply:
Fixed. The onnxItem of type `Entry<String, OnnxValue>` -> replaced with `.map(Map.Entry::getValue)`